sdk/python/tests/nem/test_TransactionFactory.py

```diff
@@ -19,73 +23,140 @@ def serialize(self):
 		# Arrange:
-		factory = TransactionFactory(FOO_NETWORK)
+		factory = TransactionFactory(Network.TESTNET)
 
 		# Act + Assert:
 		with self.assertRaises(ValueError):
-			factory.create({'type': 'multisig'})
+			factory.create({
+				'type': 'xtransfer',
+				'signer_public_key': TEST_SIGNER_PUBLIC_KEY
+			})
 
 	def test_can_create_known_transaction_with_multiple_overrides(self):
 		# Arrange:
-		factory = TransactionFactory(FOO_NETWORK, {
-			Address: lambda address: address + ' ADDRESS',
+		#  * hint on `rental_fee_recipient` is pod:Address,
+		#   type parsing rule that is getting created is for STRING 'Address'
+		# * same happens for `rental_fee`
+		#  * signer_public_key is handled differently in Transaction factory, via hint:
+		#    {'signer_public_key': PublicKey} which maps it to SDK type CryptoTypes.PublicKey
+		factory = TransactionFactory(Network.TESTNET, {
+			'Address': lambda x: x + ' but amazing',
+			'Amount': lambda _: 654321,
 			PublicKey: lambda address: address + ' PUBLICKEY'
 		})
 
 		# Act:
 		transaction = factory.create({
-			'type': 'transfer',
+			'type': 'namespace_registration',
 			'signer_public_key': 'signer_name',
-			'deadline': 98765 + 24 * 60 * 60,
-			'recipient_address': 'recipient_name',
-			'message': 'hello world',
+			'rental_fee_sink': 'fee sink',
+			'rental_fee': 'fake fee'
 		})
 
 		# Assert:
-		self.assertEqual(0x0101, transaction.type)
-		self.assertEqual(0x54000001, transaction.version)
-		self.assertEqual(98765, transaction.timestamp)
+		self.assertEqual(TransactionType.NAMESPACE_REGISTRATION, transaction.type_)
+		self.assertEqual(1, transaction.version)
+		self.assertEqual(NetworkType.TESTNET, transaction.network)
 		self.assertEqual('signer_name PUBLICKEY', transaction.signer_public_key)
 
-		self.assertEqual('recipient_name ADDRESS', transaction.recipient_address)
-		self.assertEqual(b'hello world', transaction.message)
+		self.assertEqual('fee sink but amazing', transaction.rental_fee_sink)
+		self.assertEqual(654321, transaction.rental_fee)
+
+	# endregion
+
+	# region enums handling
+
+	def test_can_create_transaction_with_default_enum_handling(self):
+		# Arrange:
+		factory = TransactionFactory(Network.TESTNET)
+
+		# Act:
+		transaction = factory.create({
+			'type': 'account_key_link',
+			'signer_public_key': 'signer_name',
+			'link_action': 'link',
+		})
+
+		# Assert:
+		self.assertEqual(LinkAction.LINK, transaction.link_action)
+
+	# endregion
+
+	# region byte array type conversion
+
+	def test_can_create_transaction_with_type_conversion(self):
+		# Arrange:
+		factory = TransactionFactory(Network.TESTNET)
+
+		# Act:
+		address = 'AEBAGBAFAYDQQCIKBMGA2DQPCAIREEYUCULBOGAB'
+		transaction = factory.create({
+			'type': 'namespace_registration',
+			'signer_public_key': 'signer_name',
+			'rental_fee_sink': Address(address)
+		})
+
+		# Assert:
+		self.assertEqual(
+			nc_Address('4145424147424146415944515143494b424d474132445150434149524545595543554c424f474142'),
```

Comment:
capitals ^^

Reply:
is there a way to force pylint to detect such things (and 0x literals?)

Reply:
idk